schematic/models/validate_attribute.py

```diff
@@ -868,186 +883,590 @@ def url_validation(
-                invalid_rows = (
-                    pd.merge(
-                        duplicated_values,
-                        missing_values,
-                        how="outer",
-                        left_index=True,
-                        right_index=True,
-                    ).index.to_numpy()
-                    + 2
-                )
-                invalid_rows = np_array_to_str_list(invalid_rows)
-                vr_errors, vr_warnings = GenerateError.generate_cross_warning(
-                    val_rule=val_rule,
-                    row_num=invalid_rows,
-                    attribute_name=source_attribute,
-                    invalid_entry=iterable_to_str_list(invalid_values.squeeze()),
-                    dmge=dmge,
+            # Concatenate target manifest columns, in a subsequent step will run cross manifest validation from
+            # the current manifest
+            # column values against the concatenated target column
+            if "value" in rule_scope:
+                target_column = ValidateAttribute._gather_target_columns_value(
+                    column_names=column_names,
+                    target_attribute=target_attribute,
+                    concatenated_target_column=target_column,
+                    target_manifest=target_manifest,
                 )
-                if vr_errors:
-                    errors.append(vr_errors)
-                if vr_warnings:
-                    warnings.append(vr_warnings)
 
-        # generate warnings if necessary
-        elif scope.__contains__("set"):
-            if (
-                val_rule.__contains__("matchAtLeastOne")
-                and len(present_manifest_log) < 1
-            ):
-                missing_entries = list(missing_manifest_log.values())
-                missing_manifest_IDs = list(missing_manifest_log.keys())
-                for missing_entry in missing_entries:
-                    missing_rows.append(missing_entry.index[0] + 2)
-                    missing_values.append(missing_entry.values[0])
+        # Store outputs according to the scope for which they are used.
+        if "set" in rule_scope:
+            validation_store = [
+                missing_manifest_log,
+                present_manifest_log,
+                repeat_manifest_log,
+            ]
 
-                missing_rows = iterable_to_str_list(set(missing_rows))
-                missing_values = iterable_to_str_list(set(missing_values))
+        elif "value" in rule_scope:
+            # From the concatenated target column, for value scope, run validation
+            (
+                missing_values,
+                duplicated_values,
+                repeat_values,
+            ) = ValidateAttribute._run_validation_across_targets_value(
+                manifest_col=manifest_col,
+                concatenated_target_column=target_column,
+            )
+            validation_store = [missing_values, duplicated_values, repeat_values]
```

Comment:
I woudl recommend validation_store being a tuple for now inseated of a list. At some point it shoudl probably be it's own class.